docs/cloud-data-warehouses/snowflake.md

```diff
@@ -1 +1,414 @@
+    4. _Monitoring_ Where you can view query history and other logs, explore various data governance stats, and review security recommendations.
+    5. _Admin_ Where you can review resource usage, and where account admins can manage warehouses, users, roles, and security policies. 
+2. Open your user menu:
+    1. Open the Develop tab in your own environment and open `transform/models/staging/stg_water_quality__stations.sql`
+    2. Click on the _Preview_ button. You should see data in the lower panel
+
+### Snowflake account structure
+Let's take a look at the hierarchy of all the different objects that make up your Snowflake project:
+![Snowflake account diagram](../../images/snowflake_account_objects.png) 
+
+### Users and roles
+In Snowflake, every user is assigned to one or more roles. Privileges are granted to roles, and this determines:
+- What databases the role can see
+- What virtual warehouses the role can use
+- What permissions the role has on database objects (e.g., read-only, read-write, etc)
+
+### Databases and schemas
+An account can have many databases, used to logically organize data.
+
+A database contains schemas:
+- Each database belongs to a single account
+- Each schema belongs to a single database
+
+A schema is a logical grouping of objects:
+- Tables, views, stages, file formats, etc
+
+### Virtual warehouses
+A virtual warehouse, usually referred to as a "warehouse" in Snowflake is a powerful and scalable computing resource that allows you to execute SQL queries and perform analytical tasks on your data. Unlike traditional data warehouses, Snowflake’s virtual warehouses are “virtual” in the sense that they are not tied to a specific hardware. Instead, they operate in the cloud and can dynamically allocate or deallocate computing resources based on your workload.
+
+A warehouse provides the required resources, such as CPU, memory, and temporary storage, to perform the following operations in a Snowflake session:
+- Executing SQL SELECT statements that require compute resources (e.g. retrieving rows from tables and views).
+- Performing DML (Data Manipulation Language) operations, such as:
+- Updating rows in tables (DELETE , INSERT , UPDATE).
+- Loading data into tables (COPY INTO <table>).
+- Unloading data from tables (COPY INTO <location>).
+
+### Warehouse sizes
+Warehouses are available in a few different sizes, depending upon the needs of the data processing job:
+
+X-small (denoted by (XS)) Good for small tasks and experimenting.
+Small (denoted by S), Suitable for single-user workloads and development.
+Medium (denoted by M), Handles moderate concurrency and data volumes.
+Large denoted by (L), Manages larger queries and higher concurrency.
+X-Large (denoted by (XL)) Powerful for demanding workloads and data-intensive operations.
+2X-Large (denoted by 2XL), 
```

Comment:
```suggestion
2X-Large (denoted by 2XL)
```

Reply:
Maybe for these last ones we can say something like: "only use these in extremely demanding circumstances where you have a strong understanding of query performance and optimization, as they can spend a lot of money quickly"